Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java

```diff
@@ -1,61 +1,206 @@
 package org.firstinspires.ftc.team6220_PowerPlay;
 
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
+
 public abstract class BaseTeleOp extends BaseOpMode {
+    double currentAngle;
+    double headingDegrees;
+    double negativeHeadingRadians;
+
+    double x;
+    double y;
+    double t;
+
+    double xRotatedVector;
+    double yRotatedVector;
+
+    double angleToCardinal;
+    double ratio;
+
     double xPower;
     double yPower;
     double tPower;
 
-    public void driveChassisWithController() {
-        xPower = gamepad1.left_stick_x * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
-        yPower = gamepad1.left_stick_y * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
-        tPower = gamepad1.right_stick_x * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
+    int slideTargetPosition = 0;
+
+    int stack = 0;
+    int[] stacks = {0, 0};
+
+    int junction = 0;
+    int[] junctions = {0, 0};
+
+    public void driveFieldCentric() {
+        currentAngle = imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES).firstAngle;
+        headingDegrees = currentAngle - startAngle;
+        negativeHeadingRadians = Math.toRadians(-headingDegrees);
+
+        if (gamepad1.left_trigger > 0.25) {
+            x = gamepad1.left_stick_x * 0.3;
+            y = -gamepad1.left_stick_y * 0.3;
+            t = gamepad1.right_stick_x * 0.2;
+        } else {
+            x = gamepad1.left_stick_x * 0.75;
+            y = -gamepad1.left_stick_y * 0.75;
+            t = gamepad1.right_stick_x * 0.5;
+        }
+
+        xRotatedVector = x * Math.cos(negativeHeadingRadians) - y * Math.sin(negativeHeadingRadians);
+        yRotatedVector = x * Math.sin(negativeHeadingRadians) + y * Math.cos(negativeHeadingRadians);
+
+        angleToCardinal = headingDegrees % 90;
+
+        if (angleToCardinal <= 10) {
+            t += angleToCardinal / 50.0;
+        } else if (angleToCardinal >= 80) {
+            t -= (90 - angleToCardinal) / 50.0;
+        }
```

Comment:
You should probably include an explanation for this in the comments

Reply:
replaced with more intuitive angle snapping